### ISSUE TITLE
Analyzer in a strong mode doesn't generate an error

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#27477]
``` dart
class C<T> {
  static const List lst = const [T];
}

main() {
  print(C.lst);
}
```

Analyzer in a strong mode produces:
No issues found

Analyzer in a nostrong mode produces:
[error] 'const' variables must be constant value 
[error] 'const' lists must have all constant values 

Strong mode should produce an error as well

Tested in dartanalyzer 1.19.1 and 1.20.0

**Comment by jmesserly** [comment]
Nice catch. This was probably exposed because we do allow type parameters to appear in constants. But naturally, in a `static` field initializer they'd be out of scope.

**Comment by jmesserly** [comment]
also worth double checking that we don't accept something like `static const List<T> list = null;`

**Comment by jmesserly** [comment]
this is still broken.

**Comment by jmesserly** [comment]
working on a fix.

**Comment by jmesserly** [comment]
https://dart-review.googlesource.com/c/sdk/+/62724

**Comment by jmesserly** [comment]
fix landed but I need to add a changelog entry

**Comment by jmesserly** [comment]
I think we can close this one now, since the fix shipped in 2.1
